### PR TITLE
462 typography consistency box/grid

### LINKED ISSUE #462: Transition to MUI - Add Consistent Typography Tags
### Important Note

Make sure to branch from [#142-MUI-migration](https://github.com/Northeastern-Electric-Racing/FinishLine/tree/%23142-MUI-migration) and request merge back to [#142-MUI-migration](https://github.com/Northeastern-Electric-Racing/FinishLine/tree/%23142-MUI-migration) for this ticket. We are doing this work on a feature branch because it affects the whole codebase and is prone to breaking things, so a feature branch allows us to test the code without breaking develop.

This ticket doesn't require changes to `package.json` or `yarn.lock`.

### Description

We want to wrap as much as possible of the text on the website with MUI `<Typography>` tags to allow the text to be up-to-date with the MUI themes. 

### Acceptance Criteria

All text in MUI `Box` and `Grid` are surrounded by `<Typography></Typography>` tags. See [WorkPackageDetails.tsx](https://github.com/Northeastern-Electric-Racing/FinishLine/blob/%23142-MUI-migration/src/frontend/src/pages/WorkPackageDetailPage/WorkPackageViewContainer/WorkPackageDetails.tsx) as an example.

### Proposed Solution

Find as much non-`Typography` text as you can and wrap them in the tags. A good place to start is by searching for all instances of MUI's `<Box>` and `<Grid>` and making sure all text inside of them is `<Typography>` and not raw HTML or raw `{ js }` strings.

You may need to `import Typography from '@mui/material/Typography'` or `import { Typography } from '@mui/material/Typography'` if it has not been done already. Be consistent with the import statement at the top of each file.

### PR DESCRIPTION
**Pull request by nezamj03**
## Changes

Typography Tags Added Where Applicable

## Notes

I have no idea how I was supposed to add the styling functions, so I just made its own folder and file and imported where needed but if I received feedback on where/how to introduce that, it would be very helpful.

## To Do

Sort of explained above, but fixing that and also I need to fix imports for consistency, but Ill do that later after the meat of the ticket is fixed.

Closes #462

### CODE REVIEW

File: src/frontend/src/pages/ChangeRequestDetailPage/StandardDetails.tsx [hunk 1]
```diff
@@ -12,31 +14,29 @@ interface StandardDetailsProps {
...
+    fontWeight: 'bold'
+  };
   return (
     <PageBlock title={'Standard Change Request Details'}>
       <Grid container spacing={1}>
         <Grid item xs={2}>
-          <Typography sx={{ maxWidth: '140px' }}>
-            <b>What</b>
-          </Typography>
+          <Typography sx={{ ...style }}>What</Typography>
```
Comment (RChandler234) [inline src/frontend/src/pages/ChangeRequestDetailPage/StandardDetails.tsx:25]: you can just do sx={style} for these

File: src/frontend/src/pages/ChangeRequestDetailPage/StandardDetails.tsx [hunk 2]
```diff
@@ -12,31 +14,29 @@ interface StandardDetailsProps {
 }
 
 const StandardDetails: React.FC<StandardDetailsProps> = ({ cr }: StandardDetailsProps) => {
+  const style = {
+    maxWidth: '140px',
+    fontWeight: 'bold'
+  };
```
Comment (RChandler234) [inline src/frontend/src/pages/ChangeRequestDetailPage/StandardDetails.tsx:20]: can you put this style object above the StandardDetails component

File: src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditDetails.tsx [hunk 3]
```diff
@@ -69,7 +69,7 @@ const WorkPackageEditDetails: React.FC<Props> = ({ users, control, errors }) =>
           />
         </Grid>
         <Grid item xs={12} md={4} sx={{ mt: 2, mb: 1 }}>
-          {statusSelect}
+          <Typography>{statusSelect}</Typography>
```
Comment (RChandler234) [inline src/frontend/src/pages/WorkPackageDetailPage/WorkPackageEditContainer/WorkPackageEditDetails.tsx:72]: the statusSelect is also a react component, so it doesn't need a Typography around it

File: src/frontend/src/styling/keyValueSameLine.tsx [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+  return (
+    <div>
+      <Typography sx={{ fontWeight: 'bold', paddingRight: padding }} display="inline">
+        {key}
+        {': '}
+      </Typography>
+      <Typography sx={{ fontWeight: 'normal', display: 'inline' }}>{value}</Typography>
+    </div>
+  );
+};
```
Comment (RChandler234) [inline src/frontend/src/styling/keyValueSameLine.tsx:17]: I like this, but it should be in the components folder, be formatted as a component, and called DetailDisplay or smthn like that

**Review by RChandler234 — CHANGES_REQUESTED** [review]

**Comment by jamescd18** [comment]
I believe this is another location where Typography is needed: https://github.com/Northeastern-Electric-Racing/FinishLine/blob/0eef41dc76b7c38e01eb92e72a12c0f7acae4d6a/src/frontend/src/pages/HomePage/UpcomingDeadlines.tsx#L46

File: src/frontend/src/pages/HomePage/UpcomingDeadlines.tsx [hunk 5]
```diff
@@ -22,6 +22,12 @@ import LoadingIndicator from '../../components/LoadingIndicator';
 import PageBlock from '../../layouts/PageBlock';
 import ErrorPage from '../ErrorPage';
 import { Grid, Typography, useTheme } from '@mui/material';
+import DetailDisplay from '../../components/DetailDisplay';
+import { DetailDisplayProps } from '../../components/DetailDisplay';
+
+const UpcomingDeadlinesDetailDisplay: React.FC<DetailDisplayProps> = ({ label, content }) => {
+  return <DetailDisplay label={label} content={content} paddingRight={2}></DetailDisplay>;
+};
```
Comment (RChandler234) [inline src/frontend/src/pages/HomePage/UpcomingDeadlines.tsx:30]: Making a whole component within the file is a bit overkill, I would just use detail display in all these places

File: src/frontend/src/pages/HomePage/WorkPackagesByTimelineStatus.tsx [hunk 6]
```diff
@@ -17,6 +17,12 @@ import LoadingIndicator from '../../components/LoadingIndicator';
 import PageBlock from '../../layouts/PageBlock';
 import ErrorPage from '../ErrorPage';
 import { FormControl, InputBase, InputLabel, MenuItem, Select, styled, Typography, useTheme } from '@mui/material';
+import DetailDisplay from '../../components/DetailDisplay';
+import { DetailDisplayProps } from '../../components/DetailDisplay';
+
+const WorkPackagesByTimelineStatusDetailDisplay: React.FC<DetailDisplayProps> = ({ label, content }) => {
+  return <DetailDisplay label={label} content={content} paddingRight={2}></DetailDisplay>;
+};
```
Comment (RChandler234) [inline src/frontend/src/pages/HomePage/WorkPackagesByTimelineStatus.tsx:25]: same comment as UpcomingDeadlines

File: src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/ProjectDetails.tsx [hunk 7]
```diff
@@ -9,7 +9,14 @@ import { datePipe, dollarsPipe, fullNamePipe, weeksPipe } from '../../../utils/p
...
 import PageBlock from '../../../layouts/PageBlock';
-import { Grid, Typography } from '@mui/material';
+import Grid from '@mui/material/Grid';
+import Typography from '@mui/material/Typography';
+import DetailDisplay from '../../../components/DetailDisplay';
+import { DetailDisplayProps } from '../../../components/DetailDisplay';
+
+const ProjectDetailsDetailDisplay: React.FC<DetailDisplayProps> = ({ label, content }) => {
+  return <DetailDisplay label={label} content={content} paddingRight={2}></DetailDisplay>;
+};
```
Comment (RChandler234) [inline src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/ProjectDetails.tsx:19]: same as above, not worth making a new component for this

File: src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/WorkPackageSummary.tsx [hunk 8]
```diff
@@ -16,6 +16,12 @@ import { routes } from '../../../utils/routes';
 import WbsStatus from '../../../components/WbsStatus';
 import Grid from '@mui/material/Grid';
 import { useTheme } from '@mui/material';
+import DetailDisplay from '../../../components/DetailDisplay';
+import { DetailDisplayProps } from '../../../components/DetailDisplay';
+
+const WorkPackageSummaryDetailDisplay: React.FC<DetailDisplayProps> = ({ label, content }) => {
+  return <DetailDisplay label={label} content={content} paddingRight={1}></DetailDisplay>;
+};
```
Comment (RChandler234) [inline src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/WorkPackageSummary.tsx:24]: same comment

File: src/frontend/src/pages/WorkPackageDetailPage/WorkPackageViewContainer/WorkPackageDetails.tsx [hunk 9]
```diff
@@ -7,7 +7,13 @@ import { WorkPackage } from 'shared';
...
 import WbsStatus from '../../../components/WbsStatus';
 import PageBlock from '../../../layouts/PageBlock';
-import { Grid, Typography } from '@mui/material';
+import { Grid } from '@mui/material';
+import DetailDisplay from '../../../components/DetailDisplay';
+import { DetailDisplayProps } from '../../../components/DetailDisplay';
+
+const WorkPackageDetailsDetailDisplay: React.FC<DetailDisplayProps> = ({ label, content }) => {
+  return <DetailDisplay label={label} content={content} paddingRight={2}></DetailDisplay>;
+};
```
Comment (RChandler234) [inline src/frontend/src/pages/WorkPackageDetailPage/WorkPackageViewContainer/WorkPackageDetails.tsx:16]: same comment as above

**Review by RChandler234 — CHANGES_REQUESTED** [review]

File: src/frontend/src/components/DetailDisplay.tsx [hunk 10]
```diff
@@ -0,0 +1,21 @@
+import Typography from '@mui/material/Typography';
```
Comment (anthonybernardi) [inline src/frontend/src/components/DetailDisplay.tsx:1]: you need the license statement for any new file you make

File: src/frontend/src/pages/HomePage/UpcomingDeadlines.tsx [hunk 11]
```diff
@@ -42,61 +48,59 @@ const UpcomingDeadlines: React.FC = () => {
...
+                sx={{ marginBottom: 2 }}
+              >
+                {wbsPipe(wp.wbsNum)} - {wp.name}
+              </Link>
+              <Box>
+                <Box>
+                  <UpcomingDeadlinesDetailDisplay
+                    label="End Date"
+                    content={datePipe(wp.endDate)}
+                  ></UpcomingDeadlinesDetailDisplay>
```
Comment (anthonybernardi) [inline src/frontend/src/pages/HomePage/UpcomingDeadlines.tsx:74]: `<div ...></div>` === `<div ... />` and you should fix this everywhere you have it

**Review by anthonybernardi — CHANGES_REQUESTED** [review]

File: src/frontend/src/pages/HomePage/UpcomingDeadlines.tsx (hunk 11 above)
Comment (nezamj03) [reply src/frontend/src/pages/HomePage/UpcomingDeadlines.tsx:74]: actually its ok replace all blessed it

File: src/frontend/src/pages/HomePage/UpcomingDeadlines.tsx [hunk 12]
```diff
@@ -42,61 +43,51 @@ const UpcomingDeadlines: React.FC = () => {
...
-                    </Typography>
-                  </Box>
+                  <Typography>
+                    {wp.expectedActivities.length} Expected Activities, {wp.deliverables.length} Deliverables
+                  </Typography>
                 </Box>
-              </CardContent>
-            </Card>
-          ))}
+              </Box>
```
Comment (RChandler234) [inline src/frontend/src/pages/HomePage/UpcomingDeadlines.tsx:86]: You don't need any of the boxes in this file

File: src/frontend/src/pages/HomePage/WorkPackagesByTimelineStatus.tsx [hunk 13]
```diff
@@ -74,33 +75,21 @@ const WorkPackagesByTimelineStatus: React.FC = () => {
...
+                  <DetailDisplay label="Engineering Lead" content={fullNamePipe(wp.projectLead)} paddingRight={2} />
                 </Box>
                 <Box>
-                  <Typography sx={{ fontWeight: 'bold', paddingRight: 2 }} display="inline">
-                    Project Manager:
-                  </Typography>
-                  <Typography display="inline">{fullNamePipe(wp.projectManager)}</Typography>
+                  <DetailDisplay label="Project Manager" content={fullNamePipe(wp.projectManager)} paddingRight={2} />
                 </Box>
                 <Box>
```
Comment (RChandler234) [inline src/frontend/src/pages/HomePage/WorkPackagesByTimelineStatus.tsx:89]: Don't need any of the boxes in this file either

File: src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/WorkPackageSummary.tsx [hunk 14]
```diff
@@ -94,8 +91,7 @@ const WorkPackageSummary: React.FC<WorkPackageSummaryProps> = ({ workPackage })
               )}
             </Grid>
             <Grid item xs={6}>
-              <Typography fontWeight="bold">Deliverables:</Typography>
-              <Typography>{deliverablesList}</Typography>
+              <DetailDisplay label="Deliverables" content={deliverablesList} paddingRight={1} />
```
Comment (RChandler234) [inline src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/WorkPackageSummary.tsx:94]: Yeah I would say a list like this is an instance where the DetailDisplay isn't applicable

File: src/frontend/src/components/DetailDisplay.tsx [hunk 15]
```diff
@@ -0,0 +1,26 @@
+/*
+ * This file is part of FinishLine by NER and licensed under GNU AGPLv3.
+ * See the LICENSE file in the repository root folder for details.
+ */
+
+import Typography from '@mui/material/Typography';
+
+interface DetailDisplayProps {
+  label: string;
+  content: string | JSX.Element | null;
```
Comment (RChandler234) [inline src/frontend/src/components/DetailDisplay.tsx:10]: just string

**Review by RChandler234 — CHANGES_REQUESTED** [review]

**Review by Ari-S-123 — CHANGES_REQUESTED** [review]
You have merge conflicts.

**Review by Ari-S-123 — DISMISSED** [review]

**Review by RChandler234 — APPROVED** [review]

**Review by anthonybernardi — APPROVED** [review]
awesome!

**Pull request merged**